src/test/scala/firrtlTests/ChirrtlSpec.scala

```diff
@@ -71,6 +71,15 @@ class ChirrtlSpec extends FirrtlFlatSpec {
       }
     }
   }
+
+  behavior of "Uniqueness"
+  CheckSpec.nonUniqueExamples.map { case (description, input) =>
```

Comment:
Super nitpicky but generally should use `foreach` over `map` when you don't care about return value.

I also stylistically prefer using `for` loops for stuff like this:
```scala
for ((description, input) <- CheckSpec.nonUniqueExamples)  { ...
```
Neither of these matter very much though

Reply:
Usually I'm pretty good with `foreach` vs. `map`. Thanks for catching that.

Stylistically, I think this makes sense for a `for` loop. There's no real structure to unpack. Do you have any rough guidelines on what is "stuff like this" vs. what isn't "stuff like this" (my own curiosity). My default is always use `map` with the only exception being when multiple generators would clearly make things easier (outer products). 

Nits (and super-nits) are totally fine. The code should be in as good a shape as we can make it. Tests have a tendency to wind up as a dumping ground of bad practices due to their write once nature. This conflicts with their utility as examples for users. Making new additions there especially good code seems critical to me.

Reply:
Yeah the "stuff like this" is super fuzzy. I think generally things that call for a multi-line `foreach`, I generally prefer `for`.  It's kind of nice to have the keyword first, but I don't really have a hard and fast rule.

Yeah, I think the "tests as examples" is something we need to keep in mind more.